cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerServiceLauncher.java

```diff
@@ -178,6 +179,10 @@ public void run() {
               twillPreparer = ((SecureTwillPreparer) twillPreparer)
                 .withIdentity(TaskWorkerTwillRunnable.class.getSimpleName(), twillUserIdentity);
             }
+            String secretName = cConf.get(Constants.Twill.Security.WORKER_SECRET_DISK_NAME);
```

Comment:
Doesn't this force us to always have a secret for worker pods in k8s?

Is the plan to always have 2 secrets available in k8s, one for system services and the other for worker services?

Reply:
The operator change (and cdap-default.xml) will cause both masters and workers to use the master secret by default. This should be backwards compatible in the event that we don't specify a separate worker secret.

Reply:
How do I launch a twill app in k8s without any secret ? Is that possible with this and operator changes?

Reply:
During creation of the twill preparer, you can omit the call to withSecretDisk and that will launch a runnable without a secret.

Reply:
Discussed offline with @wyzhang, added another configuration option which give admins the option to not mount a secret in worker pods.